tests/specs/int-simplification.k

```diff
@@ -0,0 +1,190 @@
+
+    rule E *Int A +Int B +Int C +Int D +Int F *Int A => (E +Int F) *Int A +Int B +Int C +Int D [simplification]
+
+  // ###########################################################################    
+  // div
+  // ###########################################################################    
+
+    rule A /Int 1 => A  [simplification]
+
+    // safeMath mul check c / a == b where c == a * b
+    rule (A *Int B) /Int A => B  requires A =/=Int 0 [simplification]
+
+    rule ((A *Int B) /Int C) /Int B => A /Int C  requires B =/=Int 0 [simplification]
+
+    rule (A /Int 32) *Int 32 => A  requires A modInt 32 ==Int 0 [simplification]
+
+    rule (A *Int B) /Int C <=Int D => true requires 0 <=Int A andBool 0 <=Int B andBool 0 <Int C andBool A <=Int D andBool B <=Int C [simplification]
+
+  // ###########################################################################    
+  // inc-or
+  // ###########################################################################
+
+    rule 0 |Int A => A
+    rule A |Int 0 => A
+    rule A |Int A => A
+
+  // ###########################################################################    
+  // bit-and
+  // ###########################################################################    
+
+    rule 0 &Int _ => 0
+    rule _ &Int 0 => 0
+    rule A &Int A => A
+
+  // ###########################################################################    
+  // mod
+  // ###########################################################################    
+
+    rule A modInt B => A requires 0 <=Int A andBool A <Int B [simplification]
+
+  // ###########################################################################    
+  // max, min
+  // ###########################################################################    
+
+    rule minInt(A, B) => A requires A <=Int B [simplification]
+
+  // ###########################################################################    
+  // inequality
+  // ###########################################################################    
+
+    rule A +Int B <Int A => false requires 0 <=Int B [simplification]
+
+    rule A <Int A -Int B => false requires 0 <=Int B [simplification]
+
+    rule 0 <Int 1 <<Int A => true requires 0 <=Int A [simplification]
+
+    // inequality sign normalization
+    rule          A  >Int B  => B  <Int A [simplification]
+    rule          A >=Int B  => B <=Int A [simplification]
+    rule notBool (A  <Int B) => B <=Int A [simplification]
+    rule notBool (A <=Int B) => B  <Int A [simplification]
+
+    rule 0 <=Int A *Int B => true requires 0 <=Int A andBool 0 <=Int B [simplification]
+
+    rule A -Int B +Int C <=Int D => false requires D <Int A -Int B andBool 0 <=Int C [simplification]
+
+    rule minInt(A, B) <Int C  => A <Int C orBool B <Int C  [simplification]
+
+    rule A  <Int minInt(B, C) => A  <Int B andBool A  <Int C [simplification]
+    rule A <=Int minInt(B, C) => A <=Int B andBool A <=Int C [simplification]
```

Comment:
I would prefer if these were phrased as `rule LHS => true requires RHS`, for example `rule A <Int minInt(B, C) => true requires A <Int B andBool A <Int C`. That means we don't "lose information" about the fact that this came from a `minInt`.